pkg/network/usm/sharedlibraries/ebpf.go

```diff
@@ -537,43 +550,83 @@ func (e *EbpfProgram) initPrebuilt() error {
 func sysOpenAt2Supported() bool {
 	missing, err := ddebpf.VerifyKernelFuncs("do_sys_openat2")
 
-	if err == nil && len(missing) == 0 {
-		return true
+	return err == nil && len(missing) == 0
+}
+
+// fexitSupported checks if fexit type of probe is supported on the current host.
+// It does this by creating a dummy program that attaches to the given function name, and returns true if it succeeds.
+// Method was adapted from the CWS code.
+func fexitSupported(funcName string) bool {
+	if features.HaveProgramType(ebpf.Tracing) != nil {
+		return false
 	}
 
-	kversion, err := kernel.HostVersion()
+	spec := &ebpf.ProgramSpec{
+		Type:       ebpf.Tracing,
+		AttachType: ebpf.AttachTraceFExit,
+		AttachTo:   funcName,
+		Instructions: asm.Instructions{
+			asm.LoadImm(asm.R0, 0, asm.DWord),
+			asm.Return(),
+		},
+	}
+	prog, err := ebpf.NewProgramWithOptions(spec, ebpf.ProgramOptions{
+		LogDisabled: true,
+	})
+	if err != nil {
+		return false
+	}
+	defer prog.Close()
 
+	l, err := link.AttachTracing(link.TracingOptions{
+		Program: prog,
+	})
 	if err != nil {
-		log.Error("could not determine the current kernel version. fallback to do_sys_open")
 		return false
 	}
+	defer l.Close()
 
-	return kversion >= kernel.VersionCode(5, 6, 0)
+	return true
 }
```

Comment:
this is nice, consider (in a separate PR) introduce a "self-test" function (or interface) as part of the base ebpf program struct. this can allow a better (cleaner) troubleshooting and UX when trying to use USM